crates/pixi_build_frontend/src/tool/cache.rs

```diff
@@ -287,65 +41,137 @@ pub enum ToolCacheError {
     CacheDir(miette::Report),
 }
 
-/// Describes the specification of the tool. This can be used to cache tool
-/// information.
-#[derive(Debug, Clone, Hash, Eq, PartialEq)]
-pub enum CacheableToolSpec {
-    Isolated(IsolatedToolSpec),
-    System(SystemToolSpec),
-}
-
-/// A tool that can be invoked.
-#[derive(Debug, Clone)]
-pub enum CachedTool {
-    Isolated(IsolatedTool),
-    System(SystemTool),
-}
-
-impl From<CachedTool> for Tool {
-    fn from(value: CachedTool) -> Self {
-        match value {
-            CachedTool::Isolated(tool) => Tool::Isolated(tool),
-            CachedTool::System(tool) => Tool::System(tool),
-        }
-    }
-}
-
-impl From<IsolatedTool> for CachedTool {
-    fn from(value: IsolatedTool) -> Self {
-        Self::Isolated(value)
-    }
-}
-
-impl From<SystemTool> for CachedTool {
-    fn from(value: SystemTool) -> Self {
-        Self::System(value)
-    }
-}
-
 impl ToolCache {
     /// Construct a new tool cache.
     pub fn new() -> Self {
         Self {
             cache: DashMap::default(),
         }
     }
+
+    pub async fn get_or_install_tool(
+        &self,
+        spec: IsolatedToolSpec,
+        context: &impl ToolInstaller,
```

Comment:
Perhaps this could also just be a `dyn`. The overhead of he virtual function call is completely dwarfed by the total execution time. The advantage of using `dyn` is that it's slightly faster to compile. We could probably do that in more places.